astroquery/mast/observations.py

```diff
@@ -532,10 +532,12 @@ def download_file(self, uri, *, local_path=None, base_url=None, cache=True, clou
         base_url = base_url if base_url else self._portal_api_connection.MAST_DOWNLOAD_URL
         data_url = base_url + "?uri=" + uri
 
-        # create a local file path if none is input.  Use current directory as default.
-        if not local_path:
-            filename = os.path.basename(uri)
-            local_path = os.path.join(os.path.abspath('.'), filename)
+        # parse a local file path from local_path parameter.  Use current directory as default.
+        filename = os.path.basename(uri)
+        if not local_path:  # local file path is not defined
+            local_path = filename
+        elif os.path.isdir(local_path):  # local file path is directory
```

Comment:
I'm wondering if we want to add some logic here that if the `local_path` directory doesn't exist, the directory is created? Or is that already covered elsewhere?

Reply:
I was kind of wondering about this too... right now, a directory that doesn't exist gets created as a file in the pwd since the check on line 539 returns `false`. I considered checking if the string contains a `.` character to tell if it's a file, but we would then run into issues with paths like this: `./tmp/`

Would it make more sense to check that the string contains a `.`, but that it isn't the leading character?

Reply:
I think there's some logic with `pathlib` that you could use for this!

Something like:

```
local_path = pathlib.Path(local_path)
if not local_path.exists():
        local_path.mkdir(parents=True, exist_ok=True)
```

Reply:
Sweet! Most recent version creates a `Path` object and differentiates between a file and a directory by checking `path.suffix`. If a file extension isn't found, then it's assumed that `local_path` is meant to point to a directory. If the directory does not exist, it's then created.